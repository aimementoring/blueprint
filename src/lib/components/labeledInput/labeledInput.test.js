import React from 'react';
import renderer from 'react-test-renderer';
import LabeledInput from './labeledInput.js';

describe('LabeledInput', () => {
  it('renders properly', () => {
    const tree = renderer
      .create(<LabeledInput name="testing" value="" label="Testing input" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders properly with required', () => {
    const tree = renderer
      .create(<LabeledInput name="testing" value="" label="Testing required input" required />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders properly with required and a value', () => {
    const tree = renderer
      .create(<LabeledInput name="testing" value="ABCD" label="Testing input" required />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders properly when it is disabled', () => {
    const tree = renderer
      .create(<LabeledInput name="testing" value="" label="Testing input" disabled />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders properly when type is number', () => {
    const tree = renderer
      .create(<LabeledInput name="testing" value="" label="Testing input" type="number" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders properly when different theme', () => {
    const tree = renderer
      .create(<LabeledInput name="testing" value="" label="Testing input" theme="purple" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders properly when it has an error', () => {
    const tree = renderer
      .create(
        <LabeledInput
          name="testing"
          value=""
          label="Testing input"
          theme="purple"
          error="It is a required field"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders properly with helper text', () => {
    const tree = renderer
      .create(
        <LabeledInput name="testing" label="Testing input" helperText="My awesome helper text" />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders properly with error and helper text', () => {
    const tree = renderer
      .create(
        <LabeledInput
          name="testing"
          label="Testing input"
          error="There is an error"
          helperText="My awesome helper text"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
