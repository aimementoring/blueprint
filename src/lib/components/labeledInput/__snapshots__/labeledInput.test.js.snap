// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabeledInput renders properly 1`] = `
<div
  className="theme-portal"
>
  <div
    className="wrapper"
  >
    <input
      className=" null"
      containerClassName=""
      id="testing"
      name="testing"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      for="testing"
    >
      <span
        className="fieldName"
      >
        Testing input
      </span>
    </label>
    <span
      className="errorMessage null"
    />
  </div>
</div>
`;

exports[`LabeledInput renders properly when different theme 1`] = `
<div
  className="theme-purple"
>
  <div
    className="wrapper"
  >
    <input
      className=" null"
      containerClassName=""
      id="testing"
      name="testing"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      for="testing"
    >
      <span
        className="fieldName"
      >
        Testing input
      </span>
    </label>
    <span
      className="errorMessage null"
    />
  </div>
</div>
`;

exports[`LabeledInput renders properly when it has an error 1`] = `
<div
  className="theme-purple"
>
  <div
    className="wrapper"
  >
    <input
      className=" error"
      containerClassName=""
      id="testing"
      name="testing"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      for="testing"
    >
      <span
        className="fieldName"
      >
        Testing input
      </span>
    </label>
    <span
      className="errorMessage active"
    >
      It is a required field
    </span>
  </div>
</div>
`;

exports[`LabeledInput renders properly when it is disabled 1`] = `
<div
  className="theme-portal"
>
  <div
    className="wrapper"
  >
    <input
      className=" null"
      containerClassName=""
      disabled={true}
      id="testing"
      name="testing"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      for="testing"
    >
      <span
        className="fieldName"
      >
        Testing input
      </span>
    </label>
    <span
      className="errorMessage null"
    />
  </div>
</div>
`;

exports[`LabeledInput renders properly when type is number 1`] = `
<div
  className="theme-portal"
>
  <div
    className="wrapper"
  >
    <input
      className=" null"
      containerClassName=""
      id="testing"
      name="testing"
      onChange={[Function]}
      type="number"
      value=""
    />
    <label
      for="testing"
    >
      <span
        className="fieldName"
      >
        Testing input
      </span>
    </label>
    <span
      className="errorMessage null"
    />
  </div>
</div>
`;

exports[`LabeledInput renders properly with required 1`] = `
<div
  className="theme-portal"
>
  <div
    className="wrapper"
  >
    <input
      className=" null"
      containerClassName=""
      id="testing"
      name="testing"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <label
      for="testing"
    >
      <span
        className="fieldName"
      >
        Testing required input
      </span>
    </label>
    <span
      className="errorMessage null"
    />
  </div>
</div>
`;

exports[`LabeledInput renders properly with required and a value 1`] = `
<div
  className="theme-portal"
>
  <div
    className="wrapper"
  >
    <input
      className=" null"
      containerClassName=""
      id="testing"
      name="testing"
      onChange={[Function]}
      required={true}
      type="text"
      value="ABCD"
    />
    <label
      for="testing"
    >
      <span
        className="fieldName"
      >
        Testing input
      </span>
    </label>
    <span
      className="errorMessage null"
    />
  </div>
</div>
`;
